File: php/src/Model/WorkerInviteResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Result of inviting a worker. `inviteUrl` is credential-equivalent until consumed; it is returned anyway because `emailStatus` is often `mailer_unconfigured`, which leaves sharing the link as the only delivery path.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerInviteResult
{
    public function __construct(
        public readonly PublicWorkerIdentity $identity,
        public readonly bool $workerCreated,
        /** 'sent' | 'mailer_unconfigured' | 'send_failed' */
        public readonly string $emailStatus,
        public readonly string $inviteUrl,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            identity: PublicWorkerIdentity::fromArray($data['identity'] ?? []),
            workerCreated: (bool) ($data['workerCreated'] ?? false),
            emailStatus: (string) ($data['emailStatus'] ?? ''),
            inviteUrl: (string) ($data['inviteUrl'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerInviteResult.java
package io.fivexer.sdk.model;

/**
 * Result of inviting a worker. {@code inviteUrl} is credential-equivalent until consumed; it is returned anyway because {@code emailStatus} is often {@code mailer_unconfigured}, which leaves sharing the link as the only delivery path.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerInviteResult {
    private PublicWorkerIdentity identity;
    private Boolean workerCreated;
    private String emailStatus;
    private String inviteUrl;

    public PublicWorkerIdentity getIdentity() { return identity; }
    public Boolean getWorkerCreated() { return workerCreated; }
    public String getEmailStatus() { return emailStatus; }
    public String getInviteUrl() { return inviteUrl; }
}
